### ISSUE TITLE
Uploading a file to Salesforce when in a community 

### ISSUE THREAD

**Issue by helloniklas** [forcedotcom/SalesforceMobileSDK-iOS#1957]
I'm attempting to use the 

request(forUploadFile data: Data, name: String, description: String, mimeType: String) -> SFRestRequest

It comes back with an error "The requested API is not available to portal users. You may need to specify a community for portal user access."

But there is no way to specify the community in this call like I do for other chatter api calls?

From after v36 it says I first need to upload the file to Salesforce then using that ID to post a comment etc.

**Comment by bhariharan** [comment]
@helloniklas Our base `apiUrl` is computed from `communityUrl` in a community context or from `instanceUrl` in an org context. My guess is this API is not enabled for communities by the service.

**Comment by helloniklas** [comment]
@bhariharan thanks, but I'm really struggling with this. All APIs etc are switched on and chatter works fine otherwise.

If I print the request path out that the request(forFileUploadFile has it doesn't include the community path like you mention.

```
        let uid = UUID().uuidString
        let request = SFRestAPI.sharedInstance().request(forUploadFile: imageData, name: uid, description: "Desc", mimeType: "image/jpeg")
        print(request.path)

```

Just prints /v36.0/connect/files/users/me

If I put breakpoints in to check the credentials communityUrl it has the correct community path.

Seems to the API method is not correctly building the path for the community? Or is this somehow created at a later stage? But the error coming back seems to say it's not pointing to the correct community end point?

**Comment by helloniklas** [comment]
@bhariharan  right, this must be an SDK bug. If I manually set the path after the creation of the request it now seems to work, so I do

```
       let uid = UUID().uuidString
        let request = SFRestAPI.sharedInstance().request(forUploadFile: imageData, name: uid, description: "Desc", mimeType: "image/jpeg")
        request.path = "/v36.0/connect/communities/\(Constants.communityId)/files/users/me"

```

**Comment by bhariharan** [comment]
@helloniklas You shouldn't have to manually append `communityId` to `path` for it to work. Our `apiUrl` computation takes care of prefixing the request to `communityUrl` instead of `instanceUrl`, which negates the need to append `communityId` in the `path`.

See [this](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1941) bug fix I put in. I'm guessing this is why your `baseUrl` is incorrectly computed, thereby leading to failure of the request. This should be fixed in `5.1`.
